From 924639477da1b2ef15da196d2dab6dc20ce004a3 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Sat, 13 Sep 2025 15:27:31 +0200
Subject: [PATCH 095/N] venv: also install the versioned launchers

Like on Linux we provide python3.x globally, so make
sure those also exist in the venv, so when python3.x is run
the venv version is launched and not the global one.
---
 Lib/venv/__init__.py | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/Lib/venv/__init__.py b/Lib/venv/__init__.py
index 17ee28e..4199ff5 100644
--- a/Lib/venv/__init__.py
+++ b/Lib/venv/__init__.py
@@ -388,8 +388,18 @@ class EnvBuilder:
                     f'pythonw{exe_t}{exe_d}.exe': pythonw_exe,
                 }
 
+            if sys._is_mingw:
+                for key, value in list(link_sources.items()):
+                    if key.startswith('python'):
+                        link_sources[key.replace('python', 'python3')] = value
+                        link_sources[key.replace('python', f'python3.{sys.version_info[1]}')] = value
+                for key, value in list(copy_sources.items()):
+                    if key.startswith('python'):
+                        copy_sources[key.replace('python', 'python3')] = value
+                        copy_sources[key.replace('python', f'python3.{sys.version_info[1]}')] = value
+
             do_copies = True
-            if self.symlinks:
+            if self.symlinks and not sys._is_mingw:
                 do_copies = False
                 # For symlinking, we need all the DLLs to be available alongside
                 # the executables.
